Bottom: 5a1eec8210dd8b2e4d01822be3b55da4a1811022
Top:    9b24967b175b9f3d0c0f4c6db69582e9d6fa2ea9
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-28 01:02:54 +0100

Refresh of hid-remove-unused-set_line_cap

---

diff --git a/src/hid.h b/src/hid.h
index 36aca75..2b9ca40 100644
--- a/src/hid.h
+++ b/src/hid.h
@@ -366,11 +366,6 @@ typedef enum
        assembly drawings so far. */
     void (*set_draw_faded) (hidGC gc_, int faded_);
 
-    /* When you pass the same x,y twice to draw_line, the end caps are
-       drawn as if the "line" were parallel to the line defined by these
-       coordinates.  Use this for rotating non-round pads.  */
-    void (*set_line_cap_angle) (hidGC gc_, int x1_, int y1_, int x2_, int y2_);
-
     /* The usual drawing functions.  "draw" means to use segments of the
        given width, whereas "fill" means to fill to a zero-width
        outline.  */
diff --git a/src/hid/batch/batch.c b/src/hid/batch/batch.c
index f14d75f..a32e89b 100644
--- a/src/hid/batch/batch.c
+++ b/src/hid/batch/batch.c
@@ -213,11 +213,6 @@ batch_set_draw_faded (hidGC gc, int faded)
 }
 
 static void
-batch_set_line_cap_angle (hidGC gc, int x1, int y1, int x2, int y2)
-{
-}
-
-static void
 batch_draw_line (hidGC gc, int x1, int y1, int x2, int y2)
 {
 }
@@ -365,7 +360,6 @@ hid_batch_init ()
   batch_hid.set_line_width        = batch_set_line_width;
   batch_hid.set_draw_xor          = batch_set_draw_xor;
   batch_hid.set_draw_faded        = batch_set_draw_faded;
-  batch_hid.set_line_cap_angle    = batch_set_line_cap_angle;
   batch_hid.draw_line             = batch_draw_line;
   batch_hid.draw_arc              = batch_draw_arc;
   batch_hid.draw_rect             = batch_draw_rect;
diff --git a/src/hid/common/extents.c b/src/hid/common/extents.c
index 5ac46ea..37ff538 100644
--- a/src/hid/common/extents.c
+++ b/src/hid/common/extents.c
@@ -105,11 +105,6 @@ extents_set_draw_faded (hidGC gc, int faded)
 {
 }
 
-static void
-extents_set_line_cap_angle (hidGC gc, int x1, int y1, int x2, int y2)
-{
-}
-
 #define PEX(x,w) if (box.X1 > (x)-(w)) box.X1 = (x)-(w); \
 	if (box.X2 < (x)+(w)) box.X2 = (x)+(w)
 #define PEY(y,w) if (box.Y1 > (y)-(w)) box.Y1 = (y)-(w); \
@@ -197,7 +192,6 @@ hid_extents_init (void)
   extents_hid.set_line_width      = extents_set_line_width;
   extents_hid.set_draw_xor        = extents_set_draw_xor;
   extents_hid.set_draw_faded      = extents_set_draw_faded;
-  extents_hid.set_line_cap_angle  = extents_set_line_cap_angle;
   extents_hid.draw_line           = extents_draw_line;
   extents_hid.draw_arc            = extents_draw_arc;
   extents_hid.draw_rect           = extents_draw_rect;
diff --git a/src/hid/common/hidnogui.c b/src/hid/common/hidnogui.c
index 84ceda3..cc7d2fc 100644
--- a/src/hid/common/hidnogui.c
+++ b/src/hid/common/hidnogui.c
@@ -116,12 +116,6 @@ nogui_set_draw_faded (hidGC gc, int faded)
 }
 
 static void
-nogui_set_line_cap_angle (hidGC gc, int x1, int y1, int x2, int y2)
-{
-  CRASH;
-}
-
-static void
 nogui_draw_line (hidGC gc, int x1, int y1, int x2, int y2)
 {
   CRASH;
@@ -455,7 +449,6 @@ HID hid_nogui = {
   nogui_set_line_width,
   nogui_set_draw_xor,
   nogui_set_draw_faded,
-  nogui_set_line_cap_angle,
   nogui_draw_line,
   nogui_draw_arc,
   nogui_draw_rect,
@@ -517,7 +510,6 @@ apply_default_hid (HID * d, HID * s)
   AD (set_line_cap);
   AD (set_line_width);
   AD (set_draw_xor);
-  AD (set_line_cap_angle);
   AD (draw_line);
   AD (draw_arc);
   AD (fill_circle);
diff --git a/src/hid/gcode/gcode.c b/src/hid/gcode/gcode.c
index 50f22cb..6514fe9 100644
--- a/src/hid/gcode/gcode.c
+++ b/src/hid/gcode/gcode.c
@@ -741,12 +741,6 @@ gcode_set_draw_faded (hidGC gc, int faded)
 }
 
 static void
-gcode_set_line_cap_angle (hidGC gc, int x1, int y1, int x2, int y2)
-{
-  CRASH;
-}
-
-static void
 use_gc (hidGC gc)
 {
   int need_brush = 0;
@@ -1057,7 +1051,6 @@ hid_gcode_init ()
   gcode_hid.set_line_width      = gcode_set_line_width;
   gcode_hid.set_draw_xor        = gcode_set_draw_xor;
   gcode_hid.set_draw_faded      = gcode_set_draw_faded;
-  gcode_hid.set_line_cap_angle  = gcode_set_line_cap_angle;
   gcode_hid.draw_line           = gcode_draw_line;
   gcode_hid.draw_arc            = gcode_draw_arc;
   gcode_hid.draw_rect           = gcode_draw_rect;
diff --git a/src/hid/gerber/gerber.c b/src/hid/gerber/gerber.c
index 9642579..29eeafa 100644
--- a/src/hid/gerber/gerber.c
+++ b/src/hid/gerber/gerber.c
@@ -56,7 +56,6 @@ static void gerber_set_line_cap (hidGC gc, EndCapStyle style);
 static void gerber_set_line_width (hidGC gc, int width);
 static void gerber_set_draw_xor (hidGC gc, int _xor);
 static void gerber_set_draw_faded (hidGC gc, int faded);
-static void gerber_set_line_cap_angle (hidGC gc, int x1, int y1, int x2, int y2);
 static void gerber_draw_line (hidGC gc, int x1, int y1, int x2, int y2);
 static void gerber_draw_arc (hidGC gc, int cx, int cy, int width, int height, int start_angle, int delta_angle);
 static void gerber_draw_rect (hidGC gc, int x1, int y1, int x2, int y2);
@@ -809,12 +808,6 @@ gerber_set_draw_faded (hidGC gc, int faded)
 }
 
 static void
-gerber_set_line_cap_angle (hidGC gc, int x1, int y1, int x2, int y2)
-{
-  CRASH;
-}
-
-static void
 use_gc (hidGC gc, int radius)
 {
   int c;
@@ -1194,7 +1187,6 @@ hid_gerber_init ()
   gerber_hid.set_line_width      = gerber_set_line_width;
   gerber_hid.set_draw_xor        = gerber_set_draw_xor;
   gerber_hid.set_draw_faded      = gerber_set_draw_faded;
-  gerber_hid.set_line_cap_angle  = gerber_set_line_cap_angle;
   gerber_hid.draw_line           = gerber_draw_line;
   gerber_hid.draw_arc            = gerber_draw_arc;
   gerber_hid.draw_rect           = gerber_draw_rect;
diff --git a/src/hid/gtk/gtkhid-gdk.c b/src/hid/gtk/gtkhid-gdk.c
index cfbef00..8c3af43 100644
--- a/src/hid/gtk/gtkhid-gdk.c
+++ b/src/hid/gtk/gtkhid-gdk.c
@@ -481,12 +481,6 @@ ghid_set_draw_faded (hidGC gc, int faded)
   printf ("ghid_set_draw_faded(%p,%d) -- not implemented\n", gc, faded);
 }
 
-void
-ghid_set_line_cap_angle (hidGC gc, int x1, int y1, int x2, int y2)
-{
-  printf ("ghid_set_line_cap_angle() -- not implemented\n");
-}
-
 static int
 use_gc (hidGC gc)
 {
diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index 6cb6e56..8d5e3d2 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -2124,7 +2124,6 @@ hid_gtk_init ()
   ghid_hid.set_line_width           = ghid_set_line_width;
   ghid_hid.set_draw_xor             = ghid_set_draw_xor;
   ghid_hid.set_draw_faded           = ghid_set_draw_faded;
-  ghid_hid.set_line_cap_angle       = ghid_set_line_cap_angle;
   ghid_hid.draw_line                = ghid_draw_line;
   ghid_hid.draw_arc                 = ghid_draw_arc;
   ghid_hid.draw_rect                = ghid_draw_rect;
diff --git a/src/hid/gtk/gui.h b/src/hid/gtk/gui.h
index 8a52b5a..1d5757f 100644
--- a/src/hid/gtk/gui.h
+++ b/src/hid/gtk/gui.h
@@ -480,7 +480,6 @@ void ghid_set_line_cap (hidGC gc, EndCapStyle style);
 void ghid_set_line_width (hidGC gc, int width);
 void ghid_set_draw_xor (hidGC gc, int _xor);
 void ghid_set_draw_faded (hidGC gc, int faded);
-void ghid_set_line_cap_angle (hidGC gc, int x1, int y1, int x2, int y2);
 void ghid_draw_line (hidGC gc, int x1, int y1, int x2, int y2);
 void ghid_draw_arc (hidGC gc, int cx, int cy, int xradius, int yradius,
                     int start_angle, int delta_angle);
diff --git a/src/hid/lesstif/main.c b/src/hid/lesstif/main.c
index b22328c..554bbae 100644
--- a/src/hid/lesstif/main.c
+++ b/src/hid/lesstif/main.c
@@ -3243,12 +3243,6 @@ lesstif_set_draw_faded (hidGC gc, int faded)
   /* We don't use this */
 }
 
-static void
-lesstif_set_line_cap_angle (hidGC gc, int x1, int y1, int x2, int y2)
-{
-  CRASH;
-}
-
 #define ISORT(a,b) if (a>b) { a^=b; b^=a; a^=b; }
 
 static void
@@ -3901,7 +3895,6 @@ hid_lesstif_init ()
   lesstif_hid.set_line_width          = lesstif_set_line_width;
   lesstif_hid.set_draw_xor            = lesstif_set_draw_xor;
   lesstif_hid.set_draw_faded          = lesstif_set_draw_faded;
-  lesstif_hid.set_line_cap_angle      = lesstif_set_line_cap_angle;
   lesstif_hid.draw_line               = lesstif_draw_line;
   lesstif_hid.draw_arc                = lesstif_draw_arc;
   lesstif_hid.draw_rect               = lesstif_draw_rect;
diff --git a/src/hid/nelma/nelma.c b/src/hid/nelma/nelma.c
index 7d136d6..9dbf24c 100644
--- a/src/hid/nelma/nelma.c
+++ b/src/hid/nelma/nelma.c
@@ -807,12 +807,6 @@ nelma_set_draw_faded(hidGC gc, int faded)
 }
 
 static void
-nelma_set_line_cap_angle(hidGC gc, int x1, int y1, int x2, int y2)
-{
-	CRASH;
-}
-
-static void
 use_gc(hidGC gc)
 {
 	int             need_brush = 0;
@@ -1072,7 +1066,6 @@ hid_nelma_init()
   nelma_hid.set_line_width      = nelma_set_line_width;
   nelma_hid.set_draw_xor        = nelma_set_draw_xor;
   nelma_hid.set_draw_faded      = nelma_set_draw_faded;
-  nelma_hid.set_line_cap_angle  = nelma_set_line_cap_angle;
   nelma_hid.draw_line           = nelma_draw_line;
   nelma_hid.draw_arc            = nelma_draw_arc;
   nelma_hid.draw_rect           = nelma_draw_rect;
diff --git a/src/hid/png/png.c b/src/hid/png/png.c
index cb38659..e47c641 100644
--- a/src/hid/png/png.c
+++ b/src/hid/png/png.c
@@ -1183,12 +1183,6 @@ png_set_draw_faded (hidGC gc, int faded)
 }
 
 static void
-png_set_line_cap_angle (hidGC gc, int x1, int y1, int x2, int y2)
-{
-  CRASH;
-}
-
-static void
 use_gc (hidGC gc)
 {
   int need_brush = 0;
@@ -1549,7 +1543,6 @@ hid_png_init ()
   png_hid.set_line_width      = png_set_line_width;
   png_hid.set_draw_xor        = png_set_draw_xor;
   png_hid.set_draw_faded      = png_set_draw_faded;
-  png_hid.set_line_cap_angle  = png_set_line_cap_angle;
   png_hid.draw_line           = png_draw_line;
   png_hid.draw_arc            = png_draw_arc;
   png_hid.draw_rect           = png_draw_rect;
diff --git a/src/hid/ps/eps.c b/src/hid/ps/eps.c
index afcfb73..0aca27e 100644
--- a/src/hid/ps/eps.c
+++ b/src/hid/ps/eps.c
@@ -43,7 +43,6 @@ static void eps_set_line_cap (hidGC gc, EndCapStyle style);
 static void eps_set_line_width (hidGC gc, int width);
 static void eps_set_draw_xor (hidGC gc, int _xor);
 static void eps_set_draw_faded (hidGC gc, int faded);
-static void eps_set_line_cap_angle (hidGC gc, int x1, int y1, int x2, int y2);
 static void eps_draw_rect (hidGC gc, int x1, int y1, int x2, int y2);
 static void eps_draw_line (hidGC gc, int x1, int y1, int x2, int y2);
 static void eps_draw_arc (hidGC gc, int cx, int cy, int width, int height, int start_angle, int delta_angle);
@@ -493,12 +492,6 @@ eps_set_draw_faded (hidGC gc, int faded)
 }
 
 static void
-eps_set_line_cap_angle (hidGC gc, int x1, int y1, int x2, int y2)
-{
-  CRASH;
-}
-
-static void
 use_gc (hidGC gc)
 {
   if (linewidth != gc->width)
@@ -662,7 +655,6 @@ hid_eps_init ()
   eps_hid.set_line_width      = eps_set_line_width;
   eps_hid.set_draw_xor        = eps_set_draw_xor;
   eps_hid.set_draw_faded      = eps_set_draw_faded;
-  eps_hid.set_line_cap_angle  = eps_set_line_cap_angle;
   eps_hid.draw_line           = eps_draw_line;
   eps_hid.draw_arc            = eps_draw_arc;
   eps_hid.draw_rect           = eps_draw_rect;
diff --git a/src/hid/ps/ps.c b/src/hid/ps/ps.c
index 79459d6..92f03fd 100644
--- a/src/hid/ps/ps.c
+++ b/src/hid/ps/ps.c
@@ -939,12 +939,6 @@ ps_set_draw_faded (hidGC gc, int faded)
 }
 
 static void
-ps_set_line_cap_angle (hidGC gc, int x1, int y1, int x2, int y2)
-{
-  CRASH;
-}
-
-static void
 use_gc (hidGC gc)
 {
   if (gc->me_pointer != &ps_hid)
@@ -1369,7 +1363,6 @@ hid_ps_init ()
   ps_hid.set_line_width     = ps_set_line_width;
   ps_hid.set_draw_xor       = ps_set_draw_xor;
   ps_hid.set_draw_faded     = ps_set_draw_faded;
-  ps_hid.set_line_cap_angle = ps_set_line_cap_angle;
   ps_hid.draw_line          = ps_draw_line;
   ps_hid.draw_arc           = ps_draw_arc;
   ps_hid.draw_rect          = ps_draw_rect;
